libs/dh/admin/data-access-api/src/lib/dh-admin-edit-user.store.ts

```diff
@@ -70,31 +77,55 @@ export class DhAdminEditUserStore extends ComponentStore<State> {
           }) => {
-            const requests: Observable<unknown>[] = [];
-
-            requests.push(
-              this.marketParticipantUserHttpClient.v1MarketParticipantUserUpdateUserIdentityPut(
-                userId,
-                { firstName, lastName, phoneNumber }
-              )
-            );
-
-            if (updateUserRoles) {
-              requests.push(this.userRolesStore.assignRoles(userId, updateUserRoles));
-            }
-
-            return forkJoin(requests).pipe(
-              tapResponse(
-                () => {
-                  this.setSaving(SavingState.SAVED);
-                  onSuccessFn();
+            return this.apollo
+              .mutate({
+                mutation: UpdateUserAndRolesDocument,
+                refetchQueries: [UserOverviewSearchDocument, GetUserByIdDocument],
+                variables: {
+                  updateUserInput: {
+                    userId,
+                    userIdentityUpdateDto: {
+                      firstName,
+                      lastName,
+                      phoneNumber,
+                    },
+                  },
+                  updateRolesInput: {
+                    userId,
+                    input: updateUserRoles,
+                  },
                 },
-                (error: HttpErrorResponse) => {
-                  this.setSaving(ErrorState.GENERAL_ERROR);
-                  onErrorFn(error.status, createApiErrorCollection(error));
-                }
-              )
-            );
+              })
+              .pipe(
+                tapResponse(
+                  (response) => {
+                    if (
+                      response.data?.updateUserIdentity?.success &&
+                      response.data?.updateUserIdentity.errors === null &&
+                      response.data?.updateUserRoleAssignment?.success &&
+                      response.data?.updateUserRoleAssignment.errors === null
+                    ) {
+                      this.setSaving(SavingState.SAVED);
+                      onSuccessFn();
+                    } else if (response.data?.updateUserIdentity?.errors) {
+                      this.setSaving(ErrorState.GENERAL_ERROR);
+                      onErrorFn(
+                        response.data?.updateUserIdentity.errors[0].statusCode,
+                        response.data?.updateUserIdentity.errors[0]
+                      );
+                    } else if (response.data?.updateUserRoleAssignment?.errors) {
+                      this.setSaving(ErrorState.GENERAL_ERROR);
+                      onErrorFn(
+                        response.data?.updateUserRoleAssignment.errors[0].statusCode,
+                        response.data?.updateUserRoleAssignment.errors[0]
+                      );
+                    }
+                  },
+                  () => {
+                    this.setSaving(ErrorState.GENERAL_ERROR);
+                  }
+                )
+              );
```

Comment:
suggestion: Have you considered the possibility of combining these two mutations into one (in the BFF)? There is quite a bit of complexity in determining if the mutation is successful or not. It would be nice if it was just something like this:

```ts
return this.apollo
  .mutate({
    mutation: UpdateUserDocument,
    refetchQueries: [UserOverviewSearchDocument, GetUserByIdDocument],
    variables: {
      id,
      firstName,
      lastName,
      phoneNumber,
      roles,
    },
  })
  .pipe(
    tapResponse(
      (response) => {
        if (response.data?.success)  {
          this.setSaving(SavingState.SAVED);
          onSuccessFn();
        } else if (response.data.errors) {
          this.setSaving(ErrorState.GENERAL_ERROR);
          onErrorFn(
            response.data.errors[0].statusCode,
            response.data.errors[0]
          );
        } 
      },
      () => {
        this.setSaving(ErrorState.GENERAL_ERROR);
      }
    )
  );
```

Or with the new util:

```ts
updateUser = mutation(UpdateUserDocument, {
  refetchQueries: [UserOverviewSearchDocument, GetUserByIdDocument],
});

// ...

await updateUser
  .mutate({
    input: {
      id,
      firstName,
      lastName,
      phoneNumber,
      roles,
    }
  })
  .then(response => {
    if (response.data?.success)  {
      this.setSaving(SavingState.SAVED);
      onSuccessFn();
    } else if (response.data.errors) {
      this.setSaving(ErrorState.GENERAL_ERROR);
      onErrorFn(response.data.errors[0].statusCode, response.data.errors[0]);
    } else if (response.error) {
      this.setSaving(ErrorState.GENERAL_ERROR);
    }
  },
});
```